### ISSUE TITLE
Problem upgrading Ring plugin

### ISSUE THREAD

**Issue by Venger2005** [hoobs-org/HOOBS#608]
**Description**
Hoobs notified me that there was an update available for the Ring plugin and so I tried to upgrade through the Hoobs interface.  However, I'm unable to successfully upgrade the Ring plug-in to the new version 8.0.0 which was released within the last couple of hours.  I've tried completely removing and reinstalling the plugin, with no change in outcome.  FYI - I had followed the plugin's own instructions to remove the old cameras from my Home app following the upfrade.

**Expected behavoir**
Plugin would successfully update and cameras would show in the Home app - now as bridged accessories, rather than previously needing to be added manually.

**Plugin name**
Homebridge-ring

**Please report the issue to the plugin developer**
The responses I received on the same issue which I posted on the plugin page:

_“You need to be running homebridge v1.0.0 and higher and currently are running hoobs... Install Homebridge.”_

_“Hoobs does not ship homebridge v1.0.0. They frankensteined something together to mimic the changes done in homebridge 1.0.0. Seems like they forgot something.”_

_“afaik there is still 0.4.x code running with some "adjustments" trying to match 1.0.0 behavior.”_

**Version**
Hoobs version 3.2.3

**Did you upgrade**
Please let us know if you upgraded from a previous version.
- [ ] Yes
- [X] No

**Pervious version**
N/A - problem only started after trying to update plugin.  Previous v7 of Ring plugin was working as expected on Hoobs 3.2.3

**Did you orginally upgrade to HOOBS 3 from HOOBS 2.1.1?**
Did you perform an upgrade from HOOBS 2.1.1? Some things are different.
- [X] Yes
- [ ] No

**What device are you using?**
Please let us know the device you are running HOOBS on.
- [ ] HOOBS Box
- [ ] HOOBS SD Card
- [X ] Custom Install

**If this is a custom install, what's your operating system?**
Raspbian GNU/Linux 10 (buster).

**Post your log**

```
[Ring] Configuring 4 cameras and 4 devices for location "Home" - locationId: XXXXXX-XXXX-X
5/3/2020, 11:59:17 PM [Ring] Adding new accessory lpd_v2 Front Door
5/3/2020, 11:59:17 PM [Ring] Error connecting to API
5/3/2020, 11:59:17 PM [Ring] TypeError: accessory.configureController is not a function
    at new Camera (/home/hoobs/.hoobs/node_modules/homebridge-ring/lib/homebridge/camera.js:20:19)
    at /home/hoobs/.hoobs/node_modules/homebridge-ring/lib/homebridge/ring-platform.js:188:123
    at Array.forEach (<anonymous>)
    at RingPlatform.<anonymous> (/home/hoobs/.hoobs/node_modules/homebridge-ring/lib/homebridge/ring-platform.js:167:28)
    at Generator.next (<anonymous>)
    at fulfilled (/home/hoobs/.hoobs/node_modules/homebridge-ring/lib/homebridge/ring-platform.js:5:58)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

**Comment by fLyRhyTHeMz** [comment]
exact same issue happening for me. Has made cameras unresponsive. is there a way to downgrade the plugin?

**Comment by Venger2005** [comment]
> exact same issue happening for me. Has made cameras unresponsive. is there a way to downgrade the plugin?

Yes there is - I’ve done this myself until a fix is in place. Follow the guide here: https://github.com/hoobs-org/HOOBS/wiki/4.0-Plugins#installing-plugins-manually

command is:
npm i homebridge-ring@7.2.2

Make sure you follow the guide and insert the necessary changes to your config.json

**Comment by fLyRhyTHeMz** [comment]
can you please elaborate. I don't wanna mess anything up with the rest of my system

**Comment by Venger2005** [comment]
I don’t know how you downgrade from 8.0.0 to 7.2.2 without uninstalling the plugin first. That’s what I did before I followed the guide i linked. Of course as soon as you uninstall v8 then your config is gone necessitating all the changes in the linked guide to be followed. Can’t explain it outside the scope of what I did to get mine back up and working.

**Comment by fLyRhyTHeMz** [comment]
Hopefully one of the devs can elaborate on how to do this. Actually the ability to pick versions would be a great addition to the plugin installer.

**Comment by DavidWhicker** [comment]
I’d like to see a fix here as well

**Comment by Supereg** [comment]
https://github.com/dgreif/ring/issues/308

**Comment by BobbySlope** [comment]
Heres how you can install the Ring Plugin on HOOBS
https://plugins.hoobs.org/plugin/homebridge-ring
is an Workaround while we preparing the Update, once the update is released we will certify the newer version and update the readme

**Comment by Supereg** [comment]
May I ask why my comment referencing the issue at the ring plugin got deleted? Just curious?

**Comment by Venger2005** [comment]
Upgrading to the recently-released Hoobs version 3.2.4 has allowed me to install the latest version of the Ring plugin and fixes the problem.
